src/pages/blog/post/der-komplette-guide-rund-um-sql-injections.mdx

```diff
@@ -0,0 +1,303 @@
+Vorausgesetzt, diese werden überall im Code angewendet.
+
+Prepared Statements sind SQL-Abfragen, die mit entsprechenden Parametern vorbereitet werden, aber erst zu einem späteren Zeitpunkt den eigentlichen Wert für den Parameter zugewiesen bekommen.
+Dabei wird der Wert des Parameters durch einen Placeholder markiert.
+Anbei ein Beispiel in Java:
+
+```java
+// Statement wird erzeugt
+PreparedStatement ps = connection.prepareStatement(
+    "SELECT user, password FROM tbl_user WHERE user = ?"
+);
+
+// Parameter werden übergeben
+ps.setString(1, username);
+
+// Statement wird ausgeführt.
+ResultSet rs = ps.executeQuery();
+
+```
+
+Doch was passiert da?
+
+1. Das Statement `SELECT user, password FROM tbl_user WHERE user = ?`an den Datenbank-Server gesendet und intern wird eine ID für das vorbereitete Statement gespeichert.
+2. Dem Parameter `?` (als Wert für das Feld `user`) die Zeichenkette aus der Variable `username` zugewiesen.
+3. Das Statement wird an den Datenbank-Server versendet und ausgeführt.
+
+Durch die korrekte Verwendung von Datentypen (string, int, float, ...) wird sichergestellt, dass die Werte dementsprechend escaped werden.
+Damit ist ein unerwünschter Ausbruch, zum Beispiel aus der Zeichenkette, nicht möglich.
+
+#### Prepared Statements und Performance
+
+Bei mehrfacher Ausführung desselben Statements mit unterschiedlichen Variablen kann es sogar zu einem positiven Geschwindigkeits-Impact kommen.
+Der Grund: Der Datenbank-Server kennt die Anfrage bereits und kann daraufhin die Ausführung optimieren.
+
+Bei einer einmaligen Ausführung des Queries ist die Nutzung eines Prepared Statements im Bezug auf die [Performance minimal teurer](https://orangematter.solarwinds.com/2014/11/19/analyzing-prepared-statement-performance/) (im Vergleich zu eiem einfachen SQL Statement).
+Der Grund: Die Applikation kommuniziert bei der Query-Vorbereitung bereits mit dem Datenbank-Server.
+Dies führt zu zusätzlichen Netzwerk-Roundtrips.
+
+In der Praxis ist dies jedoch selten ein Problem, da:
+1. oft andere Stellen mit höherem Optimierungsbedarf in der Applikation existieren
+2. die zusätzliche Sicherheit einen größeren Benefit liefert als die Optimierung im µs oder ms-Bereich
+
+### Aktuelle(re) Versionen
+
+Die Nutzung von aktuelle(re)n Versionen von Datenbanken oder Client-Libraries ist bevorzugt. Oft enthalten neuere Versionen notwendige Sicherheitsupdates, um bekannten Sicherheitslücken vorzubeugen oder integrieren zusätzliche Features, die die Sicherheit erhöhen können.
+
+Doch Vorsicht: Auch neuere Versionen können neue Sicherheitslücken beinhalten.
+Immer der allerneuesten Version hinterherzujagen ist ggf. auch nicht die beste Empfehlung.
+Eine gute Orientierung bieten die Lifecycle-Angaben eurer Software:
+
+* Wie lange wird eine entsprechende Version noch unterstützt und erhält sie Sicherheitsupdates?
+* Gibt es eine Long-Term-Support-Version (LTS)?
```

Comment:
hier episode zu LTS einbauen?

Reply:
Fixed in https://github.com/EngineeringKiosk/webpage/pull/231/commits/b8d02d898e59c4997ec2e64e65f4df7b28061742